src/pkg/packager/network.go

```diff
@@ -109,5 +130,154 @@ func (p *Packager) handleSgetPackage() error {
 
 	p.cfg.DeployOpts.PackagePath = localPath
 
+	spinner.Success()
 	return nil
 }
+
+func (p *Packager) handleOciPackage() error {
+	message.Debug("packager.handleOciPackage()")
+	ref, err := registry.ParseReference(strings.TrimPrefix(p.cfg.DeployOpts.PackagePath, "oci://"))
+	if err != nil {
+		return fmt.Errorf("failed to parse OCI reference: %w", err)
+	}
+
+	out := p.tmp.Base
+	message.Debugf("Pulling %s", ref.String())
+	message.Infof("Pulling Zarf package from %s", ref)
+
+	src, err := utils.NewOrasRemote(ref)
+	if err != nil {
+		return err
+	}
+
+	estimatedBytes, err := getOCIPackageSize(src, ref)
+	if err != nil {
+		return err
+	}
+
+	dst, err := file.New(out)
+	if err != nil {
+		return err
+	}
+	defer dst.Close()
+
+	// Create a thread to update a progress bar as we save the package to disk
+	doneSaving := make(chan int)
+	title := fmt.Sprintf("Pulling Zarf package data (%s of %s)", utils.ByteFormat(float64(0), 2), utils.ByteFormat(float64(estimatedBytes), 2))
+	progressBar := message.NewProgressBar(estimatedBytes, title)
+	src.ProgressBar = nil
+	go func() {
+		for {
+			select {
+			case <-doneSaving:
+				// We have been notified that we are done saving the package, no longer updating progress bar
+				title = fmt.Sprintf("Pulling Zarf package data (%s of %s)", utils.ByteFormat(float64(estimatedBytes), 2), utils.ByteFormat(float64(estimatedBytes), 2))
+				progressBar.Update(estimatedBytes, title)
+				progressBar.Successf("Pulling Zarf package data")
```

Comment:
This is very similar to the stuff done elsewhere, this pattern may be good to abstract into the progressbar file itself.